src/NuGet.Core/NuGet.Protocol.Core.Types/SourceCacheContext.cs

```diff
@@ -12,7 +12,10 @@ namespace NuGet.Protocol.Core.Types
     /// </summary>
     public class SourceCacheContext : IDisposable
     {
-        private static readonly string _tempFolderGuid = Guid.NewGuid().ToString();
```

Comment:
The negative impact of it being static was that one temp directory would be created for the process, but it would get deleted in the dispose of each instance of the class. Which can cause in use temp files to get deleted from under the method using them,